### PR TITLE
Igal/install config

### PR DESCRIPTION
**Pull request by tsorya**

### CODE REVIEW

File: render_files.py [hunk 1]
```diff
@@ -34,21 +38,24 @@ def upload_to_aws(s3_client, local_file, bucket, s3_file):
...
-            if bmh_utils.is_bmh_cr_file(file_data['path']):
-                bmh_utils.update_bmh_cr_file(file_data, hosts_list)
-
-    with open(ignition_file, "w") as file_obj:
-        json.dump(data, file_obj)
+    try:
+        if inventory_endpoint:
+            hosts_list = utils.get_inventory_hosts(inventory_endpoint, cluster_id)
+        else:
+            hosts_list = test_utils.get_test_list_hosts(cluster_id)
```
Comment (eranco74) [inline render_files.py:47]: Add log that it's using test data

**Review by yevgeny-shnaidman — COMMENTED** [review]
please reinstate the [TODO] comments that you removed. i will need then when we move to 4.6

File: render_files.py [hunk 2]
```diff
@@ -77,6 +84,38 @@ def debug_print_upload_to_s3(install_dir):
             print("Uploading file %s as object %s" % (file_path, s3_file_name))
 
 
+def install_config_manipulation(before_installer, config_dir):
```
Comment (eranco74) [inline render_files.py]: Consider changing this method to context

**Review by yevgeny-shnaidman — APPROVED** [review]

File: render_files.py [hunk 3]
```diff
@@ -77,6 +84,40 @@ def debug_print_upload_to_s3(install_dir):
...
+        subprocess.check_output(["mkdir", "-p", config_dir])
+        with open(os.path.join(config_dir, INSTALL_CONFIG), 'w+') as file_obj:
+            file_obj.write(install_config)
+    if not os.path.isdir(config_dir):
+        raise Exception('installer directory is not mounted')
+
+    if not os.path.isfile(os.path.join(config_dir, INSTALL_CONFIG)):
+        raise Exception("install config file not located in installer dir")
+
+    install_config_manipulation(config_dir=config_dir, before_installer=True)
```
Comment (eranco74) [inline render_files.py]: All these if seems irrelevant :wink: 
Consider this:
```suggestion
def prepare_install_config(config_dir, install_config):
    if not install_config:
            raise Exception("install config file not located in installer dir")
    subprocess.check_output(["mkdir", "-p", config_dir])
    install_config_file = os.path.join(config_dir, INSTALL_CONFIG)
    log.info("writing install config to file")
    with open(install_config_file, 'w+') as file_obj:
         file_obj.write(install_config)
    install_config_manipulation(config_dir=config_dir, before_installer=True)
```

Comment (tsorya) [reply render_files.py]: right , just didn't wanted to change code :) just to move it. But yes your right

Comment (tsorya) [reply render_files.py]: not sure suggested change matches current logic

File: render_files.py (hunk 2 above)
Comment (tsorya) [reply render_files.py]: thanks, nice idea

Comment (tsorya) [reply render_files.py]: forgot python :)

File: render_files.py [hunk 4]
```diff
@@ -77,6 +86,45 @@ def debug_print_upload_to_s3(install_dir):
...
+    if not os.path.exists(install_config_path):
+        logging.info("writing install config to file")
+        with open(os.path.join(config_dir, INSTALL_CONFIG), 'w+') as file_obj:
+            file_obj.write(install_config)
+
+
+def create_config_dir(work_dir):
+    config_dir = os.path.join(work_dir, "installer_dir")
+    subprocess.check_output(["mkdir", "-p", config_dir])
+    return os.path.join(work_dir, "installer_dir")
```
Comment (eranco74) [inline render_files.py]:
```suggestion
    return config_dir
```

**Review by eranco74 — APPROVED** [review]
LGTM

File: render_files.py (hunk 4 above)
Comment (tsorya) [reply render_files.py]: sorry, missed it :)
